niworkflows/anat/mni.py

```diff
@@ -214,14 +342,102 @@ def _validate_results(self):
 
 
 def mask(in_file, mask_file, new_name):
+    """
+    Applies a binary mask to an image.
+
+    Parameters
+    ----------
+    in_file : str
+              Path to a NIfTI file.
+    mask_file : str
+                Path to a NIfTI file.
+    new_name : str
+               Path/filename for the masked output image.
+
+    Returns
+    -------
+    str
+        Absolute path of the masked output image.
+
+    Notes
+    -----
+    in_file and mask_file must be in the same
+    image space and have the same dimensions.
+    """
     import nibabel as nb
     import os
-
+    # Load the input image
     in_nii = nb.load(in_file)
+    # Load the mask image
     mask_nii = nb.load(mask_file)
+    # Set all non-mask voxels in the input file to zero.
     data = in_nii.get_data()
     data[mask_nii.get_data() == 0] = 0
+    # Save the new masked image.
     new_nii = nb.Nifti1Image(data, in_nii.affine, in_nii.header)
     new_nii.to_filename(new_name)
     return os.path.abspath(new_name)
 
+
+def create_cfm(in_file, out_path, lesion_mask, global_mask=True):
+    """
+    Create a mask to constrain registration.
+
+    Parameters
+    ----------
+    in_file : str
+        Path to an existing image (usually a mask).
+        If global_mask = True, this is used as a size/dimension reference.
+    out_path : str
+        Path/filename for the new cost function mask.
+    lesion_mask : str, optional
+        Path to an existing binary lesion mask.
+    global_mask : bool
+        Create a whole-image mask (True) or limit to reference mask (False)
+
+    Returns
+    -------
+    str
+        Absolute path of the new cost function mask.
+
+    Notes
+    -----
+    in_file and lesion_mask must be in the same
+    image space and have the same dimensions
+    """
+    import os
+    import numpy as np
+    import nibabel as nb
+    
+    # Load the input image
+    in_img = nb.load(in_file)
+    in_data = in_img.get_data()
+
+    # If we want a global mask, create one based on the input image.
+    if global_mask is True:
```

Comment:
If you preemptively check for consistency of `global_mask` and `lesion_mask` (see comment on line 435), then you can make this more pythonic:

```Python
if global_mask:
```